### PR TITLE
Added additional fields for task:help printout

### PR DESCRIPTION
**Pull request by tmgerhart**
Added "notation for required or not" and also the "default setting".  This will help to make the command creation easier without having to reference back to the read.md.

### CODE REVIEW

**Review by jgamblin — DISMISSED** [review]
Looks good to me.

File: lib/helpers.rb [hunk 1]
```diff
@@ -92,8 +93,13 @@ def write_file(directory,filename,output)
       ###
       ### Helper to upload to kenna api
       ###
+<<<<<<< Updated upstream
       def upload_file_to_kenna_connector(connector_id, api_host, api_token, filename)
         # optionally upload the file if a connector ID has been specified 
+=======
```
Comment (shotop) [inline lib/helpers.rb:99]: Are these changes expected?

Comment (jgamblin) [reply lib/helpers.rb:99]: Good point... missed those when reading through this.  Thoughts @tmgerhart?

**Comment by jcran** [comment]
Concerned this will make the task listing unweildy

**Comment by tmgerhart** [comment]
Good point.  Listed below is a sample of what the help output would look
like  (Thoughts?):

dzq6k6@Kenna-Gerhart:/mnt/c/Users/toby/OneDrive/Documents/GitHub/toolkit$
docker run -it --rm toolkit task=csv2kdi:help

Running: Kenna::Toolkit::Csv2kdi

- Task Option: csv_in (string): Required? true: CSV to be converted to KDI
JSON

               (Default setting: input.csv)

- Task Option: has_header (string): Required? false: Does the input file
have a header?

               (Default setting: true)

- Task Option: meta_file (string): Required? true: File to map input to
Kenna fields

               (Default setting: meta.csv)

- Task Option: skip_autoclose (string): Required? false: If vuln not in
scan, do you want to close vulns?

               (Default setting: false)

- Task Option: assets_only (string): Required? false: Field to indicate
assets only - no vulns

               (Default setting: false)

- Task Option: domain_suffix (string): Required? false: Optional domain
suffix for hostnames

               (Default setting: )

- Task Option: input_directory (string): Required? false: Where input files
are found. Path is relative to /opt/app/toolkit/

               (Default setting: input)

- Task Option: output_directory (string): Required? false: If set, will
write a file upon completion. Path is relative to /opt/app/toolkit/

               (Default setting: output)

- Task Option: kenna_api_host (string): Required? false: Host used for the
API endpoint

               (Default setting: api.kennasecurity.com)

- Task Option: kenna_connector_id (integer): Required? false: ID required
for connector to ingest file converted

               (Default setting: )

- Task Option: kenna_api_key (string): Required? false: Kenna API code to
be used to ingest

               (Default setting: )



*Toby Gerhart **·** Kenna SE Director **· © 214-298-8882*



*From:* Jonathan Cran <notifications@github.com>
*Sent:* Monday, October 26, 2020 11:26 AM
*To:* KennaPublicSamples/toolkit <toolkit@noreply.github.com>
*Cc:* tmgerhart <toby.gerhart@kennasecurity.com>; Mention <
mention@noreply.github.com>
*Subject:* Re: [KennaPublicSamples/toolkit] Added additional fields for
task:help printout (#47)



Concerned this will make the task listing unwieldy

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/KennaPublicSamples/toolkit/pull/47#issuecomment-716663650>,
or unsubscribe

**Comment by tmgerhart** [comment]
That is exactly what it is meant for…..just the task help requested.



*Toby Gerhart **·** Kenna SE Director **· © 214-298-8882*



*From:* Jonathan Cran <notifications@github.com>
*Sent:* Monday, October 26, 2020 11:25 AM
*To:* KennaPublicSamples/toolkit <toolkit@noreply.github.com>
*Cc:* tmgerhart <toby.gerhart@kennasecurity.com>; Mention <
mention@noreply.github.com>
*Subject:* Re: [KennaPublicSamples/toolkit] Added additional fields for
task:help printout (#47)



Is the goal to print help per task? Maybe we should add a default option to
each task that prints the help only for that task?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/KennaPublicSamples/toolkit/pull/47#issuecomment-716663254>,
or unsubscribe

**Comment by tmgerhart** [comment]
Could definitely be a unconscious (unknowing) error on my part but I did
not intend for that to be updated if it was an action of mine.   I will
grab someone to discern where I went wrong.  The only lines intended to be
updated are listed below:



      def print_task_help(task_name)

        task = TaskManager.tasks.select{|x| x.metadata[:id] == task_name
}.first.new

        task.class.metadata[:options].each do |o|

          puts "- Task Option: #{o[:name]} (#{o[:type]}): Required?
#{o[:required]}: #{o[:description]}"

          puts "               (Default setting: #{o[:default]})"

        end

      end



*Toby Gerhart **·** Kenna SE Director **· © 214-298-8882*



*From:* Jerry Gamblin <notifications@github.com>
*Sent:* Monday, October 26, 2020 11:04 AM
*To:* KennaPublicSamples/toolkit <toolkit@noreply.github.com>
*Cc:* tmgerhart <toby.gerhart@kennasecurity.com>; Mention <
mention@noreply.github.com>
*Subject:* Re: [KennaPublicSamples/toolkit] Added additional fields for
task:help printout (#47)



*@jgamblin* commented on this pull request.
------------------------------

In lib/helpers.rb
<https://github.com/KennaPublicSamples/toolkit/pull/47#discussion_r512078240>
:

>        def upload_file_to_kenna_connector(connector_id, api_host, api_token, filename)

         # optionally upload the file if a connector ID has been specified

+=======

Good point... missed those when reading through this. Thoughts @tmgerhart
<https://github.com/tmgerhart>?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/KennaPublicSamples/toolkit/pull/47#discussion_r512078240>,
or unsubscribe

**Comment by tmgerhart** [comment]
Closing this to start over.

**Pull request closed without merging**